posts/2017-09-12-event-driven-serverless-app-local-dev-exp.md

```diff
@@ -0,0 +1,471 @@
+
+## Running the Application
+
+Now that we understand what the application is meant to do, and having looked at the code, let's run the application. Everything will run locally on our machine. We will [run](https://github.com/serverless/serverless/pull/4034) one service at a time, [emit](https://github.com/serverless/serverless/pull/4038) and respond to events, visualize the workflow and then call the services via `curl` simulating an application user interface.
+
+> Two new awesome commands: `serverless run` and `serverless emit` implemented as core plugins to the Serverless Framework were [released](https://github.com/serverless/serverless/blob/master/CHANGELOG.md#1200-16082017) recently.
+
+### The Local Development Experience
+
+First time you **run** any service using `serverless run`, you will get a slightly different experience. Let's run the users service, and look at the visualization on the terminal:
+
+```bash
+ $ cd mailman/services/users
+ $ sls run
+ Serverless     Installing Event Gateway
+ Serverless     Emulator initializing...
+ Serverless     Emulator listening on: http://localhost:4002
+ ...
+ Serverless     Event Gateway initializing...
+ Event Gateway  Event API listening on: http://localhost:4000
+ Event Gateway  Config API listening on: http://localhost:4001
+ Serverless     Event Gateway initialized
+ ...
+```
+The required components of the Serverless Application Platform, namely the Event Gateway and the Emulator are downloaded and installed locally on your machine, unless already installed. Also, note that the individual components advertise their API endpoints. Then, these components stay in the background listening for  events and ready for action.
+
+I want to point out that this is how the **local development experience** starts for a developer. It is all inclusive, seamless and is just there for you.
+
+### Running the Users Service
+
+With the local development setup done, we are ready to run our users service.
+
+![](https://s3-us-west-2.amazonaws.com/assets.blog.serverless.com/blog_app_sls_run_users.gif)
+
+There are a couple of things that are happening here:
+
+```bash
+ Serverless     Function 'users-register' loaded
+ Event Gateway  Function 'users-register' registered
+ Event Gateway  Subscription created for event 'http' and function 'users-register'
+```
+
+Here the Serverless Framework detects the `register` function in the `users` service and deploys it to the local Emulator.
+
+Then the `register` function from the `users` service is registered with the Event Gateway. And, a subscription is created for the `http` event and the `register` function.
+
+### Running the Email Service
+
+Let's run the email service now.
+
+![](https://s3-us-west-2.amazonaws.com/assets.blog.serverless.com/blog_app_sls_run_email.gif)
```

Comment:
This gif isn't showing for me.

Reply:
yeah, it shows up when I copied it to a browser. let's publish and see if it works, or we will fix it.